orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/SaveServiceAccountTask.java

```diff
@@ -0,0 +1,164 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Save a pipeline-scoped Fiat Service Account. The roles from
+ * this service account are used for authorization decisions when
+ * the pipeline is executed from an automated trigger.
+ */
+
+@Component
+@Slf4j
+public class SaveServiceAccountTask implements RetryableTask {
+
+  private static final String SERVICE_ACCOUNT_SUFFIX = "@managed-service-account";
+
+  @Autowired(required = false)
+  private FiatStatus fiatStatus;
+
+  @Autowired(required = false)
+  private Front50Service front50Service;
+
+  @Autowired(required = false)
+  private FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.SECONDS.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.SECONDS.toMillis(30);
+  }
+
+  @Nonnull
+  @SuppressWarnings("unchecked")
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!fiatStatus.isEnabled()) {
+      throw new UnsupportedOperationException("Fiat is not enabled, cannot save permissions.");
+    }
+
+    if (front50Service == null) {
+      throw new UnsupportedOperationException(
+        "Front50 is not enabled, no way to save pipeline. Fix this by setting front50.enabled: true");
+    }
+
+    if (!stage.getContext().containsKey("pipeline")) {
+      throw new IllegalArgumentException("pipeline context must be provided");
+    }
+
+    if (!(stage.getContext().get("pipeline") instanceof String)) {
+      throw new IllegalArgumentException(
+        "'pipeline' context key must be a base64-encoded string: Ensure you're on the most recent version of gate");
+    }
+
+    Map<String, Object> pipeline;
+    try {
+      pipeline = (Map<String, Object>) stage.decodeBase64("/pipeline", Map.class);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("pipeline must be encoded as base64", e);
+    }
+
+    if (!pipeline.containsKey("roles")) {
+      log.debug("Skipping managed service accounts since roles field is not present.");
+      return new TaskResult(ExecutionStatus.SUCCEEDED);
+    }
+
+    List<String> permissions = (List<String>) pipeline.get("roles");
+    String user = stage.getExecution().getTrigger().getUser();
+
+    if (!isUserAuthorized(user, permissions)) {
+      // TODO: Push this to the output result so Deck can show it.
+      log.warn("User {} is not authorized with all roles for pipeline", user);
+      return new TaskResult(ExecutionStatus.TERMINAL);
+    }
+
+    FiatServiceAccount svcAcct = new FiatServiceAccount(generateSvcAcctName(pipeline), permissions);
+
+    // Creating a service account with an existing name will overwrite it
+    // i.e. perform an update for our use case
+    // TODO(dibyom): If permissions are unmodified, skip the create/update below.
+    Response response = front50Service.saveServiceAccount(svcAcct);
```

Comment:
Haven't looked at Orca code in a long time, so I don't remember if there is a proper Retrofit handler when this RPC returns a 4xx/5xx status code. You may want to have a look around at other `Task`s and see if their calls to other subsystems are wrapped with try/catch blocks.